src/scripts/main.js

```diff
@@ -1,3 +1,341 @@
 'use strict';
 
-// write your code here
+const everySquare = Array.from(document.querySelectorAll('.field-cell'));
+const everyRow = Array.from(document.querySelectorAll('.field-row'));
+const mainButton = document.querySelector('.start');
+const gameScore = document.querySelector('.game-score');
+let canMoveInAnyDirection = [];
+let score = 0;
+
+function appendNumber() {
+  const emptySquares = everySquare.filter(square => square.innerHTML === '');
+
+  if (emptySquares.length === 0) {
+    return;
+  }
+
+  const index = Math.floor(Math.random() * emptySquares.length);
+  const randomSquare = emptySquares[index];
+  const value = Math.random() > 0.9 ? 4 : 2;
+
+  setCellValue(randomSquare, value);
+}
+
+function setColumn(filtered) {
+  return filtered.map(cell => parseInt(cell.innerHTML, 10) || '');
+}
+
+function setCellValue(cell, value) {
+  cell.innerHTML = value || '';
+  cell.className = 'field-cell' + (value ? ` field-cell--${value}` : '');
+}
+
+function getNewRow(column, direction) {
+  const filtered = column.filter(num => num);
+  const missing = 4 - filtered.length;
+  const unfilled = Array(missing).fill('');
+
+  return (
+    direction === 'down' || direction === 'right'
+      ? unfilled.concat(filtered)
+      : filtered.concat(unfilled)
+  );
+}
+
+function checkMoveAndFinish() {
+  if (canMoveInAnyDirection.includes(true)) {
+    canMoveInAnyDirection = [];
+    appendNumber();
+  }
+
+  checkIfGameFinished();
+}
+
+function moveDown() {
+  for (let i = 0; i < everyRow.length; i++) {
+    const filteredColumn = everySquare.filter(td => td.cellIndex === i);
+
+    moveAndCombine('down', filteredColumn);
+  };
+
+  checkMoveAndFinish();
+}
+
+function moveUp() {
+  for (let i = 0; i < everyRow.length; i++) {
+    const filteredColumn = everySquare.filter(td => td.cellIndex === i);
+
+    moveAndCombine('up', filteredColumn);
+  };
+
+  checkMoveAndFinish();
+}
```

Comment:
- Please don't forget about the DRY - don't repeat yourself
- looks like some code is duplicated in the moveDown and moveUp, so try to improve it 
- the same in a moveRight and moveLeft